src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBSpark.scala

```diff
@@ -149,9 +150,50 @@ object CosmosDBSpark extends CosmosDBLoggingTrait {
       case _: Throwable => // no op
     }
 
-    logInfo("Write config: " + writeConfig.toString)
+    val maxMiniBatchImportSizeKB: Int = writeConfig
+      .get[String](CosmosDBConfig.MaxMiniBatchImportSizeKB)
+      .getOrElse(CosmosDBConfig.DefaultMaxMiniBatchImportSizeKB.toString)
+      .toInt
+
+    val writeThroughputBudget : Option[Int] = writeConfig
+      .get[String](CosmosDBConfig.WriteThroughputBudget)
+      .map(_.toInt)
+
+    var writeThroughputBudgetPerCosmosPartition: Option[Int] = None
+    var baseMaxMiniBatchImportSizeKB: Int = maxMiniBatchImportSizeKB
 
-    val mapRdd = rdd.mapPartitions(savePartition(_, writeConfig, numPartitions), preservesPartitioning = true)
+    // if writeThroughputBudget is provided in config, derive baseMaxMiniBatchImportSize based on #sparkPartitions & baseMiniBatchRUConsumption value
+    if (writeThroughputBudget.exists(_ > 0)) {
+      val baseMiniBatchRUConsumption: Int = writeConfig
```

Comment:
Do we need to expose this config? I cant really see how customers can guesstimate this value - is this just as a backdoor for us to tweak workloads if engaged?

Reply:
good point. yes, it was meant to be a tweaking knob for us if needed in case of uncommon payload and not for users.